File: C++ Programs/Queue Programs/Queue-Using-Array-CPP-main/Queue.cpp
#include<iostream>
using namespace std;
class Queue{
	private:
		int arr[5];
		int front=-1;
		int rear=-1;
		public:
		Queue(){
			front=0;
			rear=0;
		}
		void insert(){
			if(isFull()){
				cout<<"Queue is Full!"<<endl;
				return;
			}
			else{
				cout<<"Enter value:"; 
				cin>>arr[rear];
				rear++;
				cout<<"Data entered successfully!"<<endl;
			}
		}
		void deletion(){
			if(isEmpty()){
				cout<<"Stack is Free!"<<endl;
				front=-1;
				rear=-1;
				return;
			}
			else{
				cout<<"Data removed: "<<arr[front]<<endl;
				front++;
			}	
		}
		bool isFull(){
			return rear == 5;
		}
		bool isEmpty(){
			return front == rear;
		}
		void display(){
			if(isEmpty()){
				cout<<"Stack is Free!"<<endl;
				return;
			}
			else{
				for(int i=front; i <rear; i++){
					cout<<arr[i]<<"	";
				}
				cout<<endl;
			}
		}
};
int main()
{
	Queue q1;
	int choice;
	do{
		cout<<"1. Insert Data"<<endl;
		cout<<"2. Delete Data"<<endl;
		cout<<"3. Display Data"<<endl;
		cout<<"4. Exit"<<endl;
		cout<<"Enter his choice: ";
		cin>>choice;
		switch(choice){
			case 1:
				q1.insert();
				break;
				case 2:
					q1.deletion();
					break;
					case 3:
						q1.display();
						break;
						case 4:
							exit(0);
							default:
								cout<<"You entered wrong choice!"<<endl;
		}
	} while(true);
}

File: Java Programs/Queue Programs/Queue-Using-Array-Java-main/Queue.java
import java.util.Scanner;
class Que{
    int[] arr= new int[5];
    private int front=-1;
    private int rear=-1;
    Que(){
        front=0;
    }
    public boolean isFull(){
        return rear == 5-1;
    }
    public boolean isEmpty(){
        return front == rear;
    }
    public void Enquue(){
        if(isFull()){
            System.out.println("Queue is Full!");
            return;
        }
        Scanner sc= new Scanner(System.in);
        int value;
        System.out.print("Enter value: ");
        value= sc.nextInt();
        arr[++rear]=value;
        System.out.println("value added successfully!");
    }
    public void dequeue(){
        if(isEmpty()){
            System.out.println("Queue is Free!");
            return;
        }
        else{
            System.out.println("value removed: "+arr[front]);
            front++;
        }
    }
    public void Display(){
        for(int i=front; i<=rear; i++){
            System.out.print(arr[i]+"   ");
        }
        System.out.println();
    }
}
public class Queue {
    public static void main(String[] args) {
        Que q1= new Que();
        do{
            Scanner sc= new Scanner(System.in);
            int choice;
            System.out.println("1. Insert value");
            System.out.println("2. Delete value");
            System.out.println("3. Display Queue");
            System.out.println("4. Exiting.....");
            System.out.print("Enter his choice: ");
            choice=sc.nextInt();
            switch(choice){
                case 1:
                    q1.Enquue();
                    break;
                    case 2:
                        q1.dequeue();
                        break;
                        case 3:
                            q1.Display();
                            break;
                            case 4:
                                System.exit(0);
                                default:
                                    System.out.println("You entered wrong choice!");
            }
        } while(true);
    }
}
